### PR TITLE
Add more get options (include_tree_info, include_proof, include_version)

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Add include_tree_info, include_proof, include_version flags to get options
- Merge toJsObjectShallow() to toJsObject()
- Use StateInfoProperties to generate meta data 
- Do some code refactoring

Related issues: https://github.com/ainblockchain/ain-blockchain/issues/505

### CODE REVIEW

File: common/constants.js [hunk 1]
```diff
@@ -327,11 +327,20 @@ const ProofProperties = {
...
 const StateInfoProperties = {
+  NUM_PARENTS: 'num_parents',
+  PROOF_HASH: 'proof_hash',
+  VERSION: 'version',
   TREE_HEIGHT: 'tree_height',
   TREE_SIZE: 'tree_size',
   TREE_BYTES: 'tree_bytes',
 };
 
+const GET_OPTIONS_INCLUDE_ALL = {
```
Comment (liayoo) [inline common/constants.js:338]: Is this only used for testing purposes? If so, how about moving this to test-util?

File: common/common-util.js [hunk 2]
```diff
@@ -174,6 +174,34 @@ class CommonUtil {
     }
   }
 
+  static toGetOptions(args) {
```
Comment (liayoo) [inline common/common-util.js:177]: suggestion) how about we have a separate `toMatchOptions()` as it only needs `is_global`?

File: common/constants.js (hunk 1 above)
Comment (platfowner) [reply common/constants.js]: Right, just moved.

File: common/common-util.js (hunk 2 above)
Comment (platfowner) [reply common/common-util.js:177]: Added toMatchOrEvalOptions(). Thanks for the nice suggestion!

**Comment by platfowner** [comment]
PTAL at the revised parts. @liayoo

File: common/constants.js [hunk 3]
```diff
@@ -663,7 +666,7 @@ function getGenesisSharding() {
   const config = getGenesisConfig('genesis_sharding.json');
   if (config[ShardingProperties.SHARDING_PROTOCOL] === ShardingProtocols.POA) {
     const ownerAddress = CommonUtil.getJsObject(
-        GenesisAccounts, [AccountProperties.OWNER, AccountProperties.ADDRESS]);
+        GenesisAccounts, [AccountProperties.OWNER, Accountroperties.ADDRESS]);
```
Comment (liayoo) [inline common/constants.js:669]: typo?

**Review by liayoo — APPROVED** [review]
LGTM with a comment. Thanks!!

File: common/constants.js (hunk 3 above)
Comment (platfowner) [reply common/constants.js]: Oops, fixed!

**Comment by platfowner** [comment]
Thanks for the fast review, now merging..

**Pull request merged**